syntax = "proto3";

option go_package = "github.com/llamerada-jp/colonio-seed/pkg/seed/core";

import "core/core.proto";

package colonio.NodeAccessorProtocol;

message Carrier {
  repeated Packet packet = 1;
}

message Head {
  core.NodeID dst_nid = 1;
  core.NodeID src_nid = 2;
  uint32 hop_count = 3;
  uint32 mode = 4;
  uint32 channel = 5;
  uint32 module_channel = 6;
  uint32 command_id = 7;
}

message Packet {
  // Enable head if index is 0.
  Head head = 1;
  // ID is common to sequence of packets.
  uint32 id = 2;
  // index decreses by 1 in a sequance of packets.
  uint32 index = 3;
  bytes content = 4;
}

message ICE {
  core.NodeID local_nid = 1;
  core.NodeID remote_nid = 2;
  string ice = 3;
}

message Offer {
  core.NodeID prime_nid = 1;
  core.NodeID second_nid = 2;
  string sdp = 3;
  uint32 type = 4;
}

message OfferSuccess {
  uint32 status = 1;
  core.NodeID second_nid = 2;
  string sdp = 3;
}

message OfferFailure {
  uint32 status = 1;
  core.NodeID prime_nid = 2;
}
